src/Microsoft.ServiceFabric.Powershell.Http/YamlUtils.cs

```diff
@@ -0,0 +1,100 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Powershell.Http
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Newtonsoft.Json;
+    using SfSbzYamlMergeUtils;
+
+    /// <summary>
+    /// Utility and extension methods for YAML/JSON operations.
+    /// </summary>
+    internal static class YamlUtils
+    {
+        internal enum ResourseType
+        {
+            VOLUME,
+            NETWORK,
+            APPLICATION,
+            GATEWAY,
+            UNKNOWN,
+        }
+
+        internal static List<Tuple<ResourseType, string, string, string>> GetFinalResourceDescriptionFromYamls(List<string> yamlInputFileList, string outputRootDirectory)
+        {
+            GenerateMergedResourceDescription(yamlInputFileList, outputRootDirectory);
+
+            // Get allfiles SortedbyName from the folder
+            var generatedFilesList = Directory.GetFiles(outputRootDirectory, "*.json", SearchOption.AllDirectories).Select(fn => new FileInfo(fn)).OrderBy(f => f.Name);
+
+            var resourceDescriptionList = new List<Tuple<ResourseType, string, string, string>>();
+            foreach (var fileIter in generatedFilesList)
+            {
+                // generate a list of tuple<name, type, description string>
+                // call get resource description from file
+                resourceDescriptionList.Add(GetResourceDescriptionFromFile(fileIter.FullName));
+            }
+
+            return resourceDescriptionList;
+        }
+
+        /// <summary>
+        /// Returns the Resource description from a JSON file
+        /// </summary>
+        /// <param name="jsonFilePath">the path of the json</param>
+        /// <returns>Resource description</returns>
+        internal static Tuple<ResourseType, string, string, string> GetResourceDescriptionFromFile(string jsonFilePath)
+        {
+            ResourseType rtype = ResourseType.UNKNOWN;
+            string rName = null, rDesc = null, rapiVersion = null;
+
+            // TODO Serialize with a json definition obj ??
+            using (StreamReader r = new StreamReader(jsonFilePath))
```

Comment:
Once you have a rEsourceInformation class, file ca content can be deserialized directly using JsonConvert..  [https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonConvert_DeserializeObject__1.htm](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonConvert_DeserializeObject__1.htm)